src/background_thread.c

```diff
@@ -153,6 +153,26 @@ set_current_thread_affinity(int cpu) {
 /* Minimal sleep interval 100 ms. */
 #define BACKGROUND_THREAD_MIN_INTERVAL_NS (BILLION / 10)
 
+static int
+background_thread_cond_wait(background_thread_info_t *info,
+    struct timespec *ts) {
+	int ret;
+
+	/*
+	 * pthread_cond_wait drops and re-acquires the mutex internally, w/o
+	 * going through our wrapper.  Update the locked state explicitly.
+	 */
+	atomic_store_b(&info->mtx.locked, false, ATOMIC_RELAXED);
```

Comment:
When is the `locked` used?

Reply:
It's not used in this sequence (cond_wait), but because it's locking-unlocking w/o updating the flag in our wrapper, it's causing confusion for other threads, and will trigger assertions (that's how it's found).